### ISSUE TITLE
HOME shortcut not working correctly on scrcpy v1.15

### ISSUE THREAD

**Issue by zequip** [Genymobile/scrcpy#1643]
- [x] I have read the [FAQ](https://github.com/Genymobile/scrcpy/blob/master/FAQ.md).
 - [x] I have searched in existing [issues](https://github.com/Genymobile/scrcpy/issues).

 - OS: windows 10
 - scrcpy version:  1.15
 - device model: Samsung  galaxy j7
 - Android version: Android 8.1.0

All desktop shortcuts stopped working on the latest scrcpy release (V1.15), only left and right click are recognized. If I rollback to V1.14 everything works as intended.

EDIT: remaining issue: only the <kbd>HOME</kbd> shortcut is sometimes broken https://github.com/Genymobile/scrcpy/issues/1643#issuecomment-670233833

**Comment by rom1v** [comment]
v1.15 now forwards <kbd>Ctrl</kbd> (and <kbd>Shift</kbd>) to the device, so the shortcut modifier (<kbd>MOD</kbd>) had to be changed. By default, it's <kbd>Alt</kbd> or <kbd>Super</kbd>. See https://github.com/Genymobile/scrcpy/releases/tag/v1.15

#1642

**Comment by zequip** [comment]
I see, I read the changelog but I didn't realize the change. My bad...
I'm still having problems with middle click (home button). It works fine for a couple of seconds after startup, then it stops detecting it.

**Comment by rom1v** [comment]
Oh, you're not the only one 😕 https://github.com/Genymobile/scrcpy/pull/1598#issuecomment-662218703

**Comment by rom1v** [comment]
Did you have the same problem on v1.14?

**Comment by zequip** [comment]
No problem on  v1.14.

**Comment by rom1v** [comment]
That's very weird, I could not reproduce.

Could you add this log, please:

```
diff --git a/app/src/input_manager.c b/app/src/input_manager.c
index 02dd9cb5..cf557716 100644
--- a/app/src/input_manager.c
+++ b/app/src/input_manager.c
@@ -582,6 +582,7 @@ input_manager_process_mouse_button(struct input_manager *im,
                                    const SDL_MouseButtonEvent *event) {
     bool control = im->control;
 
+    LOGD("mouse button type=%d, which=%d, button=%d", event->type, event->which, event->button);
     if (event->which == SDL_TOUCH_MOUSEID) {
         // simulated from touch events, so it's a duplicate
         return;
```

And when middle click fails, please post the line it prints in the console.

cc @AzureRaptor

**Comment by zequip** [comment]
I'm sorry... Where should I add that?

**Comment by rom1v** [comment]
OK, here is a build:
 - [`scrcpy.exe`](https://tmp.rom1v.com/scrcpy/1643/1/scrcpy.exe)
  SHA256: 97437b46f4e540e9aa51ede53fdc130615af71de74668b85d3bd53850e9ec621

(replace it in your v1.15 release)

**Comment by zequip** [comment]
This one didn't work from the start. I noticed it does work on the lockscreen but once the device is awake it stops detecting it (Alt-H isn't working either)

I'm sorry I can't be more helpful with logs and stuff, I'm not very experienced

**Comment by rom1v** [comment]
> I'm sorry I can't be more helpful with logs and stuff

Oh, I put a debug log, so it is not printed by default (should have added an info log for simplicity).

Could you please run with `scrcpy -Vdebug`?

Press <kbd>Windows</kbd>+<kbd>r</kbd>, type `cmd` and press Enter.
In the terminal, type `cd C:\Users\...(your path to scrcpy)...`
Then `scrcpy -Vdebug`.

It should print logs when you press middle-click. Copy-paste them here.

**Comment by rom1v** [comment]
Nevermind, here is a build with an INFO log:

 - [`scrcpy.exe`](https://tmp.rom1v.com/scrcpy/1643/2/scrcpy.exe)
  _SHA256: d79c39debe2306471f9f4625c77f2694c4b402380dc628cea3ea9e739a0cec7a_

(replace it in your v1.15 release)

**Comment by rom1v** [comment]
> once the device is awake it stops detecting it (Alt-H isn't working either)

When HOME stop working, what if you execute `adb shell input keyevent HOME`?

Press <kbd>Windows</kbd>+<kbd>r</kbd>, type `cmd` and press <kbd>Enter</kbd>.
In the terminal, type `cd C:\Users\...(your path to scrcpy)...`
Then type `adb shell input keyevent HOME` and press <kbd>Enter</kbd>.

**Comment by zequip** [comment]
the log says:

INFO: mouse button type=1025, which=0, button=2 (when I press middle-click)
INFO: mouse button type=1026, which=0, button=2 (when I release middle-click)

adb shell input keyevent HOME worked

**Comment by zequip** [comment]
Just in case I wasn't clear... the command worked at sending the home button, it didn't however fix the problem.

Also, the latest adb packaged with scrcpy fails at connecting wirelessly, is doesn't seem to be related to scrcpy because it works if I replace it with the previous version of adb.

**Comment by rom1v** [comment]
Oh, I think I found a possible cause for the broken HOME shortcut. Here is a fix: https://github.com/Genymobile/scrcpy/commit/8fb01922a3fda87a7fa31d7279d8f7faee890f8f

 - [`scrcpy.exe`](https://tmp.rom1v.com/scrcpy/1643/3/scrcpy.exe)
  _SHA256: b91b1f560c60d7ee8504e5cf2fa6d3766394c21dcfa45605aaef90e7e69c6fd6_

Does it fix the problem for you? @zequip @AzureRaptor

**Comment by zequip** [comment]
Fixed!!

**Comment by rom1v** [comment]
Thank you. I will release a new version 1.15.1 soon then.

**Comment by rom1v** [comment]
> the latest adb packaged with scrcpy fails at connecting wirelessly

I tested this morning on Windows, it works for me.

**Comment by zequip** [comment]
no, thank YOU!! for the fix and for the amazing app.

I let you decide if the thread should be left open or closed.

**Comment by zequip** [comment]
> I tested this morning on Windows, it works for me.

I might be having some overlapping issue with the previous adb, then‽ It's weird because it works just fine while connected through USB...
I tried deleting `.android` folder on `C:\Users\username` but this wasn't enough. Do you know if there's any other location where some other configuration file might be hiding?

**Comment by zequip** [comment]
Disregard everything I said about ADB. I don't know exactly what I did but it's working now.

**Comment by rom1v** [comment]
Fixed in v1.15.1.
